### PR TITLE
Check for a valid godep version.

### PR DESCRIPTION
**Pull request by david-mcmahon**

### CODE REVIEW

**Comment by thockin** [comment]
I don't think we want to demand that EVERYONE has godep installed.  Most people will never need it.  I think we can only check it in {update,verify}-godep-licenses.sh and godep-save.sh.  Basically anywhere we invoke godep...

**Comment by david-mcmahon** [comment]
PTAL.

cc @ixdy @spxtr - anything special to do here for CI?

**Comment by ixdy** [comment]
@david-mcmahon the unit/integration build should automatically run the verify checks, so if it passes we should be good.

**Comment by david-mcmahon** [comment]
@thockin ping.

File: hack/update-godep-licenses.sh [hunk 1]
```diff
@@ -108,6 +108,7 @@ process_content () {
 #############################################################################
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
+kube::golang::verify_godep_version
```
Comment (thockin) [inline hack/update-godep-licenses.sh]: this script doesn't actually call godep - do we need to check this here?

File: hack/verify-godep-licenses.sh [hunk 2]
```diff
@@ -20,6 +20,7 @@ set -o pipefail
 
 KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
+kube::golang::verify_godep_version
```
Comment (thockin) [inline hack/verify-godep-licenses.sh]: this script doesn't actually call godep - do we need to check this here?

**Comment by thockin** [comment]
I'll LGTM but I think 2 of 3 are superfluous.

File: hack/update-godep-licenses.sh (hunk 1 above)
Comment (david-mcmahon) [inline hack/update-godep-licenses.sh]: Correct.  Thanks for catching.  Same as below.

**Comment by david-mcmahon** [comment]
Updated. PTAL.

**Comment by thockin** [comment]
@matchstick FYI

**Comment by matchstick** [comment]
Is this really better than just vendoring godep?
I don't feel strongly either way, but then we know that the one in the source base works. 

mrubin

**Pull request merged**